### PR TITLE
make key type generic. preserve validation of key type to accept only Long and UUID.

### PR DESCRIPTION
**Pull request by razvanmuscalu**

### CODE REVIEW

File: src/main/java/tech/picnic/jolo/LoaderFactoryBuilderImpl.java [hunk 1]
```diff
@@ -64,60 +64,62 @@
...
     return relation(left, right).manyToMany(leftKey, rightKey);
   }
 
-  private static <L extends Record, R extends Record> TableField<?, Long> getForeignKeySymmetric(
+  @SuppressWarnings("unchecked")
+  private static <L extends Record, R extends Record, K> TableField<?, K> getForeignKeySymmetric(
       Table<L> left, Table<R> right) {
-    TableField<?, Long> leftKey = getOptionalForeignKey(right, left).orElse(null);
-    TableField<?, Long> rightKey =
+    TableField<?, K> leftKey = (TableField<?, K>) getOptionalForeignKey(right, left).orElse(null);
```
Comment (Mordavolt) [inline src/main/java/tech/picnic/jolo/LoaderFactoryBuilderImpl.java]: You can avoid casting here and elsewhere by doing 
```suggestion
    TableField<?, K> leftKey = Util.<R, L, K>getOptionalForeignKey(right, left).orElse(null);
```

**Review by Mordavolt — COMMENTED** [review]
Hey, thanks for your contribution! I didn't have time for a full review, but one thing you can certainly improve is the explicit casting, please check out the suggestion.

File: src/main/java/tech/picnic/jolo/LoaderFactoryBuilderImpl.java (hunk 1 above)
Comment (razvanmuscalu) [reply src/main/java/tech/picnic/jolo/LoaderFactoryBuilderImpl.java]: Thanks for the suggestion, I now removed the explicit castings I introduced by using this `Util` method.

**Comment by razvanmuscalu** [comment]
BTW tests pass for both `openjdk8` and `openjdk11`, but the build for `openjdk8` fails because of connection error to sonar on `localhost/127.0.0.1:9000`, not sure what's going on.

**Comment by Stephan202** [comment]
The Sonar failure is due to:

> Encrypted environment variables have been removed for security reasons.

As a result `SONARCLOUD_TOKEN` isn't set. This happens because your PR comes from an external repository. Would have to look into best practices on how to solve this. :thinking:

**Pull request closed without merging**